hooks/useGas.ts

```diff
@@ -46,10 +66,35 @@ const useGas = () => {
 		}
 	}, []);
 
+	useEffect(() => {
+		const maxPriorityFeePerGas = selectedGas.maxPriorityFeePerGas;
+		const maxFeePerGasValue = isCustomGasPrice ? gasPriceWei : selectedGas.maxFeePerGas;
+
+		const l1GasConfig = { maxPriorityFeePerGas, maxFeePerGas: maxFeePerGasValue };
+		const l2GasConfig = { gasPrice: gasPriceWei };
+
+		const config = isL2 ? l2GasConfig : l1GasConfig;
+
+		setGasConfig(config as GasPrice);
```

Comment:
Just confirming this GasPrice casting was only because config is only a partial match with GasPrice. If so, maybe we should make a GasConfig type more suited to kwenta with all fields optional.

Reply:
yes thats correct - its a partial of GasPrice - I will update this now - good idea

Reply:
so I typed this out - but im still finding it hard to get this done without casting as the `GasConfig` type. not sure if you can guide me on typescript at all for this scenario

Reply:
if this still works, the solution you added above is sufficient